app/main.py

```diff
@@ -1,8 +1,35 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    result = []
+    for person in people:
+        person_object = Person(person["name"], person["age"])\
+            if person["name"] not in Person.people \
+            else Person.people[person["name"]]
+        result.append(person_object)
+
+    for person in people:
+        wife_husband = "wife" if "wife" in person else "husband"
```

Comment:
It's better to check for both wife and husband explicitly instead of assuming one if the other is not present.